motr/idx_dix.c

```diff
@@ -991,7 +1002,7 @@ static void dix_index_create_ast(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	dix.dd_layout.dl_type = DIX_LTYPE_DESCR;
```

Comment:
As I said in previous comment, the "dix" is output of function dix_build().
Why do we update the "dix.dd_layout.dl_type" forcibly.
If we do it forcibly, then what is the purpose of dix_build()?

Reply:
I would like to explain requirement. Each client request is sending two RPC calls. First RPC call to get layout type and pool version. Second RPC call to create relevant operation.
 
**Main Goal:** To avoid first RPC call

dix_build() is common function for all index and meta data operations due to that reason selected for changes.
We are trying to get pool version and layout type in dix_discovery() depends on layout type. It will not process if layout type is DIX_LTYPE_DESCR. Due to that reason we are assigned forcibly and assigned pool version and hash values.